### PR TITLE
Deprecate AudioStreamCallback. Update comments to make behaviour clearer

### PR DESCRIPTION
**Pull request by dturner**

### CODE REVIEW

File: include/oboe/AudioStreamCallback.h [hunk 1]
```diff
@@ -108,26 +109,29 @@ class AudioStreamErrorCallback {
...
+     * In that case, the app MUST stop() and close() the stream.
      *
-     * Note that this method will be called on a thread created by Oboe.
+     * This method will be called on a thread created by Oboe.
      *
      * @param audioStream pointer to the associated stream
      * @param error
-     * @return true if the error has been handled, false if not
+     * @return true if the stream has been stopped and closed,
+     * false if not
```
Comment (philburk) [inline include/oboe/AudioStreamCallback.h:126]: Will this get picked up by Doxygen?

File: include/oboe/AudioStreamCallback.h [hunk 2]
```diff
@@ -108,26 +109,29 @@ class AudioStreamErrorCallback {
...
-     * then the stream will be closed and onErrorAfterClose() will be closed.
+     * If this method returns false it indicates that the stream has *not been stopped and closed
+     * by the application. In this case it will be stopped by Oboe in the following way:
+     * onErrorBeforeClose() will be called, then the stream will be closed and onErrorAfterClose()
+     * will be closed.
      *
-     * If this method returns true then the normal error processing will not occur.
-     * In that case, the app MUST stop() and close() the stream!
+     * If this method returns true it indicates that the stream *has* been stopped and closed
+     * by the application and the normal error processing will *not* occur.
```
Comment (philburk) [inline include/oboe/AudioStreamCallback.h:118]: > normal error processing will *not* occur.

Sounds abnormal. Scary.

Maybe instead "Oboe will not need to do that."

**Review by philburk — COMMENTED** [review]
Looks good but a minor nit.

File: include/oboe/AudioStreamCallback.h (hunk 1 above)
Comment (dturner) [reply include/oboe/AudioStreamCallback.h:126]: Moved back onto single line to be on the safe side.

**Pull request merged**